iosApp/SurveyTests/Sources/Specs/Presentation/Modules/SurveySelection/Account/AccountViewDataSourceSpec.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  AccountViewDataSourceSpec.swift
+//  Survey
+//
+//  Created by Bliss on 22/12/22.
+//  Copyright © 2022 Nimble. All rights reserved.
+//
+
+import Nimble
+import Quick
+import Shared
+
+@testable import Survey
+
+final class AccountViewDataSourceSpec: QuickSpec {
+
+    override func spec() {
+
+        var logOutUseCase: LogOutUseCaseKMMMock!
+        var coordinator: AccountCoordinatorMock!
+        var accountViewModel: AccountViewModel!
+        var dataSource: AccountView.DataSource!
+        var accountUiModel: AccountUiModel!
+
+        describe("a Account View View Data Source") {
+
+            beforeEach {
+                accountUiModel = AccountUiModel(imageUrl: "", name: "", appVersion: "")
+                logOutUseCase = LogOutUseCaseKMMMock()
+                coordinator = AccountCoordinatorMock()
+                accountViewModel = AccountViewModel(
+                    accountUiModel: accountUiModel,
+                    logOutUseCase: logOutUseCase
+                )
+                dataSource = .init(
+                    coordinator: coordinator,
+                    viewModel: accountViewModel
+                )
+            }
+
+            describe("its init") {
+
+                it("loading state is false") {
+                    let isLoading = try self.awaitPublisher(dataSource.$showingLoading.collectNext(1)).last
+                    expect(isLoading) == false
+                }
+
+                it("account state has correct value") {
+                    let viewState = try self.awaitPublisher(dataSource.$viewState.collectNext(1)).last
+                    expect(viewState?.accountUiModel) == accountUiModel
+                }
+            }
+
+            describe("its logout") {
+
+                beforeEach {
+                    logOutUseCase.invokeReturnValue = AnyFlow(result: KotlinUnit())
+                    delayLogout()
+                }
+
+                it("sets loading state") {
+                    let viewState = try self.awaitPublisher(dataSource.$viewState.collectNext(2)).last
+                    expect(viewState?.isLoading) == true
+                }
+
+                it("sets logout to true") {
+                    let viewState = try self.awaitPublisher(dataSource.$viewState.collectNext(3)).last
+                    expect(viewState?.isLogout) == true
+                }
+
+                it("call coordinator to show login") {
+                    _ = try self.awaitPublisher(dataSource.$viewState.collectNext(3)).last
+                    expect(coordinator.showLoginCallsCount) == 1
+                }
+            }
+        }
+
+        func delayLogout() {
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.01) {
+                dataSource.logOut()
+            }
+        }
+    }
```

Comment:
I faced issues when flow emitting new values right away and does not emit prior values. For example this series: `no loading -> loading -> no loading`, because the emit is so fast it sometimes drop to `loading`: `no loading -> no loading`. Resulted in random failing tests (two runs fail differently). 

I put this func to the end of the file to keep the format similar to other DataSource tests.